### DISCUSSION TITLE
Battery available energy exceeds maximum

### DISCUSSION THREAD

**Discussion by warmexx** [WillCodeForCats/solaredge-modbus-multi#671] [General]
### Describe the bug

My B1 Available Energy sensor has some data for about a day length. I'm not sure, but I think the data was gathered when I just moved from the "other integration" to yours. 

To debug I dumped the values for the sensor:
```
float_to_hex(self._platform.decoded_model["B_Energy_Available"]) = 0x46189c29
hex(SunSpecNotImpl.FLOAT32) = 0x7fc00000
self._platform.decoded_model['B_Energy_Available'] = 9767.0400390625
self._platform.decoded_common['B_RatedEnergy'] * self._platform.battery_rating_adjust = 9700.0
```
so it's obvious why the sensor returns no data. 

### Expected behavior

I expect to have data for this sensor.

### Screenshots

![image](https://github.com/user-attachments/assets/73ae468d-86e4-4451-8b89-8bcdf96920a0)

### Diagnostic File

[config_entry-solaredge_modbus_multi-01J6Y11X43DPTYCN1G2GSCMXTE.json](https://github.com/user-attachments/files/16931733/config_entry-solaredge_modbus_multi-01J6Y11X43DPTYCN1G2GSCMXTE.json)

### Home Assistant Version

2024.9.1

### solaredge-modbus-multi Version

2.4.20

### Installation Type

HAOS

### Read the Instructions

- [X] My configuration follows https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Configuration
- [X] I have read the Known Issues https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Known-Issues
- [X] This issue is not about a template error (do not report template errors, read https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Template-Design-Notes)

**Comment by WillCodeForCats** [comment]
Did you change the rating adjust value?

**Comment by WillCodeForCats** [comment]
https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Configuration#battery-options

**Reply by warmexx** [reply]
No I didn't, I just added your integration after disabling the other one. I did try to power cycle battery and inverter but that didn't make a difference. At the moment my battery is at 32% and I still don't have available energy. I'll set the adjust value to 1% and see what happens. Thanks for responding so quickly.

**Reply by WillCodeForCats** [reply]
I do a lot more data validation and checks because I hate seeing bad data or spikes messing up my graphs. In some cases, like this one, SolarEdge may over-rate some batteries in software above the nameplate rating.

**Comment by warmexx** [comment]
![image](https://github.com/user-attachments/assets/86aae212-9e70-410a-8936-442dab255f04)

**Reply by warmexx** [reply]
I would have expected something like available = 30% of maximum
